compiler/rustc_error_codes/src/error_codes/E0212.md

```diff
@@ -0,0 +1,52 @@
+An associated type cannot be extracted from
+a higher-ranked trait bound in the context.
+
+Erroneous code example:
+
+```compile_fail,E0212
+pub trait Foo<T> {
+    type A;
+
+    fn get(&self, t: T) -> Self::A;
+}
+
+fn foo2<I : for<'x> Foo<&'x isize>>(
+    field: I::A) {} // error!
+
+
+struct SomeStruct<I: for<'x> Foo<&'x isize>> {
```

Comment:
Maybe only have one example? These all look basically the same and I think it's more important to be concise than exhaustive here. I don't feel strongly about that though.

Reply:
I took these example from corresponding test cases. I agree with you that the important point is conciseness rather than exhaustiveness. Also, I want to know what other reviewers think about this.